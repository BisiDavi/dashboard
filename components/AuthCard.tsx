import { PropsWithChildren } from "react";
import { Card, Grid, Typography } from "@material-ui/core";
import CardContent from "@material-ui/core/CardContent";
import { authCardStyles } from "@styles/AuthCard.style";

export default function AuthCard({
    content,
    children,
}: PropsWithChildren<AuthCardProps>) {
    const classes = authCardStyles();
    return (
        <Grid container>
            <Grid className={classes.cardGrid} item lg={6} md={6}>
                <Card className={classes.card}>
                    <CardContent>
                        <div className={classes.header}>
                            <div>
                                <Typography component="h1">
                                    {content.title}
                                </Typography>
                                <Typography component="p">
                                    {content.text}
                                </Typography>
                            </div>
                        </div>
                        <div className={classes.authform}>{children}</div>
                    </CardContent>
                </Card>
            </Grid>
        </Grid>
    );
}

interface AuthCardProps {
    content: {
        title: string;
        text: string;
    };
}
