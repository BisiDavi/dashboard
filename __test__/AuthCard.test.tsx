/**
 * @jest-environment jsdom
 */

import { render } from "@testing-library/react";
import AuthCard from "@components/AuthCard";
import authJson from "@json/auth.json";

describe("Testing the AuthCard Component", () => {
    it("AUthCard Snapshot testing", () => {
        expect(
            render(<AuthCard content={authJson.login}></AuthCard>),
        ).toMatchSnapshot();
    });

    it("testing auth card props 'children'", () => {
        expect(
            render(
                <AuthCard content={authJson.login}>
                    <h1>Login page</h1>
                </AuthCard>,
            ),
        ).toBeTruthy();
    });
});
