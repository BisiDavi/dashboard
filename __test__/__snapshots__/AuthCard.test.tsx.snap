// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the AuthCard Component AUthCard Snapshot testing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiGrid-root MuiGrid-container"
      >
        <div
          class="MuiGrid-root makeStyles-cardGrid-4 MuiGrid-item MuiGrid-grid-md-6 MuiGrid-grid-lg-6"
        >
          <div
            class="MuiPaper-root MuiCard-root makeStyles-card-1 MuiPaper-elevation1 MuiPaper-rounded"
          >
            <div
              class="MuiCardContent-root"
            >
              <div
                class="makeStyles-header-2"
              >
                <div>
                  <h1
                    class="MuiTypography-root MuiTypography-body1"
                  >
                    Sign in
                  </h1>
                  <p
                    class="MuiTypography-root MuiTypography-body1"
                  >
                    Welcome, sign in with either github or google to access the dashboard
                  </p>
                </div>
              </div>
              <div
                class="makeStyles-authform-3"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiGrid-root MuiGrid-container"
    >
      <div
        class="MuiGrid-root makeStyles-cardGrid-4 MuiGrid-item MuiGrid-grid-md-6 MuiGrid-grid-lg-6"
      >
        <div
          class="MuiPaper-root MuiCard-root makeStyles-card-1 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            class="MuiCardContent-root"
          >
            <div
              class="makeStyles-header-2"
            >
              <div>
                <h1
                  class="MuiTypography-root MuiTypography-body1"
                >
                  Sign in
                </h1>
                <p
                  class="MuiTypography-root MuiTypography-body1"
                >
                  Welcome, sign in with either github or google to access the dashboard
                </p>
              </div>
            </div>
            <div
              class="makeStyles-authform-3"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
